Resources/Storage/StorageBase.py

```diff
@@ -317,17 +324,20 @@ def constructURLFromLFN( self, lfn, withWSUrl = False ):
 
       return S_ERROR( 'LFN does not follow the DIRAC naming convention %s' % lfn )
 
-    result = self.getURLBase( withWSUrl = withWSUrl )
-    if not result['OK']:
-      return result
-    urlBase = result['Value']
-    url = os.path.join( urlBase, lfn.lstrip( '/' ) )
-    return S_OK( url )
+    urlDict = dict( self.protocolParameters )
+    urlDict['Options'] = '&'.join( "%s=%s" % ( optionName, urlDict[paramName] )
+                                   for paramName, optionName in self.DYNAMIC_OPTIONS.iteritems()
+                                    if urlDict.get( paramName ) )
```

Comment:
Really ? How would you want it indented ? 
The for loop is inside the join, so I thought it makes sense to allign it this way. How should it be ?